plexpy/notifiers.py

```diff
@@ -2754,32 +2723,82 @@ class HIPCHAT(object):
     def __init__(self):
         self.apiurl = plexpy.CONFIG.HIPCHAT_URL
         self.color = plexpy.CONFIG.HIPCHAT_COLOR
-        self.incl_subject = plexpy.CONFIG.HIPCHAT_INCL_SUBJECT
         self.emoticon = plexpy.CONFIG.HIPCHAT_EMOTICON
+        self.from_subject = plexpy.CONFIG.HIPCHAT_FROM_SUBJECT
+        self.incl_pmslink = plexpy.CONFIG.FACEBOOK_INCL_PMSLINK
+        self.incl_poster = plexpy.CONFIG.FACEBOOK_INCL_POSTER
+        self.incl_subject = plexpy.CONFIG.HIPCHAT_INCL_SUBJECT
 
-    def notify(self, message, subject):
+    def notify(self, message, subject, **kwargs):
         if not message or not subject:
             return
-        
+
+        data = {'notify': 'false'}
+
+        text = message.encode('utf-8')
+
         if self.incl_subject:
-            text = subject.encode('utf-8') + ': ' + message.encode('utf-8')
-        else:
-            text = message.encode('utf-8')
+            text = subject.encode('utf-8') + ': ' + text
 
         if self.emoticon:
             text = self.emoticon + ' ' + text
 
-        data = {'message': text,
-                'notify': 'false',
-                'message_format': 'text'}
-
         if self.color:
             data['color'] = self.color
 
+        if self.from_subject:
+            data['from'] = subject.encode('utf-8')
+
+        if self.incl_poster and 'metadata' in kwargs:
+            pretty_metadata = PrettyMetadata(kwargs['metadata'])
+            poster_url = pretty_metadata.get_poster_url()
+            poster_link = pretty_metadata.get_poster_link()
+            caption = pretty_metadata.get_caption()
+            title = pretty_metadata.get_title()
+            subtitle = pretty_metadata.get_subtitle()
+            plex_url = pretty_metadata.get_plex_url()
+
+            card = {'title': title,
+                    'format': 'medium',
+                    'style': 'application',
+                    'id': '41c354f0-69a5-4251-a61c-c6a8daf8e2f6'}
```

Comment:
I used random GUID generation.  It is a required field, but seems to have no effect that I can discern.  The api docs say:

> An id that will help HipChat recognise the same card when it is sent multiple times
> Valid length range: 1 - unlimited.